server/routes/users.js

```diff
@@ -53,20 +53,25 @@ router.put(
             return;
         }
 
-        const user = await usersController.sanatize(
-            await usersController.update(id, req.body)
-        );
+        let chefProfile = {};
 
-        if (req.body.isChef && !user.isChef) {
+        if (req.body.newChef && !req.body.isChef) {
+            const user = await usersController.findOneWithId(id);
+            console.log(user);
             const data = {
                 cuisineSpecialty: [...req.body.cuisineSpecialty],
-                userId: id,
+                userId: user._id,
             };
-            const chefData = await chefsController.create(data);
-            delete chefData.userId;
-            user.chefProfile = chefData;
+            chefProfile = await chefsController.create(data);
+            delete chefProfile.userId;
+            req.body.isChef = !req.body.isChef;
```

Comment:
Unless this is actually something you can toggle back and forth, I'd rather it be more explicit (`.isChef = true`)